modules/charon-core/src/main/java/org/wso2/charon3/core/config/ExtensionBuilder.java

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.charon3.core.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.schema.AttributeSchema;
+import org.wso2.charon3.core.schema.SCIMAttributeSchema;
+import org.wso2.charon3.core.schema.SCIMDefinitions;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Abstract class for extension builder.
+ */
+public abstract class ExtensionBuilder {
+
+    /**
+     * Get rootURI of the schema.
+     *
+     * @return
+     */
+    public abstract String getURI();
+
+    /**
+     * Returns subattribute URI.
+     *
+     * @param config           ExtensionAttributeSchemaConfig.
+     * @param subAttributeName SubAttributeName.
+     * @return Subattribute URI
+     */
+    protected String getSubAttributeURI(ExtensionAttributeSchemaConfig config, String subAttributeName) {
+
+        if (isRootConfig(config)) {
+            return config.getURI() + ":" + subAttributeName;
+        } else {
+            return config.getURI() + "." + subAttributeName;
+        }
+    }
+
+    /**
+     * Returns true if it is the rootconfig. Else returns false.
+     *
+     * @param config ExtensionAttributeSchemaConfig
+     * @return Returns true if it is the rootconfig. Else returns false.
+     */
+    protected boolean isRootConfig(ExtensionAttributeSchemaConfig config) {
+
+        return getURI().equals(config.getURI());
+    }
+
+    /*
+     * Knows how to build a complex attribute
```

Comment:
Should be a doc comment.

Reply:
fixed https://github.com/wso2/charon/pull/323/commits/6a773e3384a5db3baab94370f9f64a904834a48d